server/controllers/wazuh-api.js

```diff
@@ -406,6 +427,21 @@ export class WazuhApiCtrl {
             { idHost: req.payload.id }
           );
 
+          // Check the run_as for the API user and update it
+          let apiUserAllowRunAs = false;
+          if(apiAvailable.run_as){
+            const responseApiUserAllowRunAs = await this.apiInterceptor.request(
+              'GET',
+              `${apiAvailable.url}:${apiAvailable.port}/security/users/me`,
+              {},
+              { idHost: req.payload.id }
+            );
+            if(responseApiUserAllowRunAs.status === 200) {
+              apiUserAllowRunAs = responseApiUserAllowRunAs.data.data.affected_items[0].allow_run_as;
+            }
+          }
+          CacheInMemoryAPIUserAllowRunAs.set(req.payload.id, apiAvailable.username, apiUserAllowRunAs);
```

Comment:
maybe both algorithms to check the apiUserAllowRunAs can be merged into a generic function, to avoid duplicate code.

Reply:
I extracted the code to a function to be reusable